File: BOJ/2503_1.cpp
// 숫자 야구
// https://www.acmicpc.net/problem/2503
#include <iostream>
#include <string>
using namespace std;
int TC, answer;
const int MAX = 1000;
bool check[MAX];
void init()
{
    for (int i = 123; i <= 999; i++)
    {
        string tmp = to_string(i);
        if (tmp[0] == tmp[1] || tmp[1] == tmp[2] || tmp[2] == tmp[0]) // 중복 제거
            continue;
        else if (tmp[0] == '0' || tmp[1] == '0' || tmp[2] == '0')
            continue;
        check[i] = true;
    }
}
int main()
{
    ios::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);

    init();
    cin >> TC;
    for (int ts = 0; ts < TC; ts++)
    {
        string point;
        int strike, ball;
        cin >> point >> strike >> ball;
        for (int i = 123; i <= 999; i++)
        {
            // 모든 경우의 수 검색
            string now = to_string(i);
            int strikeCnt = 0, ballCnt = 0;
            // 비교
            for (int j = 0; j < 3; j++)
            {
                for (int k = 0; k < 3; k++)
                {
                    // 위치, 숫자가 같은 경우
                    if (j == k && now[j] == point[k])
                        strikeCnt++;
                    // 숫자가 같은 경우
                    if (j != k && now[j] == point[k])
                        ballCnt++;
                }
            }

            if (strike != strikeCnt || ball != ballCnt)
                check[i] = false;
        }
    }

    for (int i = 123; i <= 999; i++)
        if (check[i])
            answer++;
    cout << answer;

    return 0;
}
/*
    참고
    https://kiung9085.tistory.com/19
*/

File: BOJ/2503_1.py
'''
    숫자 야구
    https://www.acmicpc.net/problem/2503
'''
from sys import stdin


def init():
    for i in range(123, MAX):
        now = str(i)
        if now[0] == now[1] or now[1] == now[2] or now[2] == now[0]:
            continue
        elif now[0] == "0" or now[1] == "0" or now[2] == "0":
            continue
        check[i] = True


MAX = 1000
check = [False for i in range(MAX)]

init()

questionCnt = int(stdin.readline())
for _ in range(questionCnt):
    quesion = list(map(str, stdin.readline().strip().split()))
    point = quesion[0]
    strike = int(quesion[1])
    ball = int(quesion[2])
    for i in range(123, MAX):
        now = str(i)
        strikeCnt = 0
        ballCnt = 0
        for y in range(3):
            for x in range(3):
                if y == x and now[y] == point[x]:
                    strikeCnt += 1
                if y != x and now[y] == point[x]:
                    ballCnt += 1
        if strike != strikeCnt or ball != ballCnt:
            check[i] = False
print(check.count(True))
